OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseTabsDashboardFragment.java

```diff
@@ -267,7 +410,7 @@ public void setupToolbar() {
         binding.toolbar.collapsedToolbarDismiss.setOnClickListener(v -> requireActivity().finish());
         binding.toolbar.expandedToolbarDismiss.setOnClickListener(v -> requireActivity().finish());
 
-        if (courseData.isUpgradeable() && environment.getAppFeaturesPrefs().isValuePropEnabled()) {
+        if (!isErrorCase && courseData.isUpgradeable() && environment.getAppFeaturesPrefs().isValuePropEnabled()) {
```

Comment:
I guess we can remove this button as we now using new UI

Reply:
Also no use for IAP flow is needed on CourseOutlineScreen

Reply:
this block of code is used when the user has course access and can upgrade to unlock the gated content.